### PR TITLE
analog: Fix substitution so it can find its language files

### PR DESCRIPTION
**Pull request by p-h**
###### Motivation for this change
Program crashed on startup

###### Things done
fixed sed command

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 133540` at b9eccbf5 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>analog</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/tools/admin/analog/default.nix:21:3:
```
   |
21 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl2` is a deprecated license, please check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/tools/admin/analog/default.nix:33:5:
```
   |
33 |     license = lib.licenses.gpl2;
   |     ^
```
</li>
  </ul>
</details>

File: pkgs/tools/admin/analog/default.nix [hunk 1]
```diff
@@ -14,7 +14,7 @@ stdenv.mkDerivation rec {
   postPatch = ''
     sed -i src/anlghead.h \
       -e "s|#define DEFAULTCONFIGFILE .*|#define DEFAULTCONFIGFILE \"$out/etc/analog.cfg\"|g" \
-      -e "s|#define LANGDIR .*|#define LANGDIR \"$out/share/$pname}/lang/\"|g"
+      -e "s|#define LANGDIR .*|#define LANGDIR \"$out/share/$pname/lang/\"|g"
```
Comment (SuperSandro2000) [inline pkgs/tools/admin/analog/default.nix:17]:
```suggestion
      -e "s|#define LANGDIR .*|#define LANGDIR \"$out/share/${pname}/lang/\"|g"
```

Comment (Artturin) [reply pkgs/tools/admin/analog/default.nix:17]: $name/${name} includes the version `something-0.0.1`
$pname/${pname} doesn't `something`

$pname is a shell variable
${pname} is replaced by nix before the phase is run

${} is easier to keep track of visually

**Comment by Artturin** [comment]
```
./results/analog/bin/analog
./results/analog/bin/analog: Analog CE version 6.0.17/Unix
./results/analog/bin/analog: Warning F: Failed to open logfile
  /nix/store/fn8a00136yifi83qmg90kqngp25kwkfv-analog-6.0.17/bin/logfile.log:
  ignoring it
  (For help on all errors and warnings, see docs/errors.html)
./results/analog/bin/analog: Fatal error: failed to open output file
  /nix/store/fn8a00136yifi83qmg90kqngp25kwkfv-analog-6.0.17/bin/Report.html for
  writing: exiting
```

**Pull request merged**